mapie/quantile_regression.py

```diff
@@ -0,0 +1,486 @@
+from __future__ import annotations
+from typing import Optional, Tuple, Union, cast, List
+
+import numpy as np
+from sklearn.base import RegressorMixin, clone
+from sklearn.linear_model import QuantileRegressor
+from sklearn.pipeline import Pipeline
+from sklearn.utils.validation import (
+    indexable,
+    check_is_fitted,
+    _num_samples,
+    _check_y,
+)
+from typing_extensions import TypedDict
+
+from ._typing import ArrayLike, NDArray
+from .utils import (
+    check_alpha_and_n_samples,
+    check_n_features_in,
+    check_null_weight,
+    fit_estimator,
+    check_lower_upper_bounds,
+)
+from ._compatibility import np_quantile
+from .regression import MapieRegressor
+
+
+class MapieQuantileRegressor(MapieRegressor):
+    """
+    This class implements the conformalized quantile regression strategy
+    as proposed by Romano et al. (2019) to make conformal predictions.
+    The only valid ``method`` is "quantile" and the only valid default
+    ``cv`` is "split".
+
+    Parameters
+    ----------
+    estimator : Optional[RegressorMixin]
+        Any regressor with scikit-learn API
+        (i.e. with fit and predict methods), by default ``None``.
+        If ``None``, estimator defaults to a ``QuantileRegressor`` instance.
+
+    method: str
+        Method to choose for prediction, in this case, the only valid method
+        is the "quantile" method.
+
+    cv: Optional[str]
+        The cross-validation strategy for computing residuals. The "split" cv
+        is the method used by default. It splits the dataset into a training
```

Comment:
It does not do the split since the user needs to do it before calling MAPIE.

Reply:
done